The condition should be checking for <= 11 since the change is available
in LLVM12:
https://github.com/llvm/llvm-project/commit/322d0afd875df66b36e4810a2b95c20a8f22ab9b

Upstream-Status: Submitted
Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>

diff --git a/IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp b/IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp
index 363d622c2..bbb04be30 100644
--- a/IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp
+++ b/IGC/VectorCompiler/lib/GenXCodeGen/GenXSimdCFRegion.cpp
@@ -395,7 +395,7 @@ public:
         return GenXIntrinsic::getGenXIntrinsicID(V) ==
                    GenXIntrinsic::genx_all ||
                CI->getIntrinsicID() ==
-#if LLVM_VERSION_MAJOR <= 12
+#if LLVM_VERSION_MAJOR <= 11
                   Intrinsic::experimental_vector_reduce_and;
 #else
                   Intrinsic::vector_reduce_and;
@@ -408,7 +408,7 @@ public:
       if (const CallInst *CI = dyn_cast<CallInst>(V))
         return GenXIntrinsic::getGenXIntrinsicID(V) ==
                    GenXIntrinsic::genx_any || CI->getIntrinsicID() ==
-#if LLVM_VERSION_MAJOR <= 12
+#if LLVM_VERSION_MAJOR <= 11
                    Intrinsic::experimental_vector_reduce_or;
 #else
                    Intrinsic::vector_reduce_or;
